lib/reporting/sp_proofing_events_by_uuid.rb

```diff
@@ -0,0 +1,226 @@
+                sum(name = "IdV: enter verify by mail code submitted" and properties.event_properties.success and !properties.event_properties.pending_in_person_enrollment and !properties.event_properties.fraud_check_failed) > 0 as verified_by_mail,
+                sum(name = "Fraud: Profile review passed" and properties.event_properties.success) > 0 as verified_fraud_review,
+                max(profile_age_in_seconds) as out_of_band_verification_pending_seconds,
+                sum(name = "User registration: agency handoff visited" and properties.event_properties.ial2) > 0 as agency_handoff,
+                sum(name = "SP redirect initiated" and properties.event_properties.ial == 2) > 0 as sp_redirect
+                toMillis(min(@timestamp)) as first_event
+                by properties.user_id as login_uuid
+        | filter workflow_started > 0 or verified_by_mail > 0 or verified_fraud_review > 0
+        | limit 10000
+        | sort first_event asc
+      QUERY
+      return base_query if after_row.nil?
+
+      base_query + " | filter first_event > #{after_row['first_event']}"
+    end
+
+    def as_csv
+      csv = []
+      csv << ['Date Range', "#{time_range.begin.to_date} - #{time_range.end.to_date}"]
+      csv << csv_header
+      data.each do |result_row|
+        csv << result_row
+      end
+      csv.compact
+    end
+
+    def to_csv
+      CSV.generate do |csv|
+        as_csv.each do |row|
+          csv << row
+        end
+      end
+    end
+
+    def as_emailable_reports
+      [
+        EmailableReport.new(
+          title: "#{agency_abbreviation} Proofing Events By UUID",
+          table: as_csv,
+          filename: "#{agency_abbreviation.downcase}_proofing_events_by_uuid",
+        ),
+      ]
+    end
+
+    def csv_header
+      [
+        'UUID',
+        'Workflow Started',
+        'Documnet Capture Started',
+        'Document Captured',
+        'Selfie Captured',
+        'Document Authentication Passed',
+        'SSN Submitted',
+        'Personal Information Submitted',
+        'Personal Information Verified',
+        'Phone Submitted',
+        'Phone Verified',
+        'Verification Workflow Complete',
+        'Identity Verified for In-Band Users',
+        'Identity Verified for Verify-By-Mail Users',
+        'Identity Verified for Fraud Review Users',
+        'Out-of-Band Verification Pending Seconds',
+        'Agency Handoff Visited',
+        'Agency Handoff Submitted',
+      ]
+    end
+
+    def data
+      return @data if defined? @data
+
+      login_uuid_data ||= fetch_results.map do |result_row|
+        process_result_row(result_row)
+      end
+      login_uuid_to_agency_uuid_map = build_uuid_map(login_uuid_data.map(&:first))
+
+      @data = login_uuid_data.map do |row|
+        login_uuid, *row_data = row
+        agency_uuid = login_uuid_to_agency_uuid_map[login_uuid]
+        next if agency_uuid.nil?
+        [agency_uuid, *row_data]
```

Comment:
this will leave out deleted users, right? should we put like some sort of deleted placeholder?

Reply:
I believe that for this report we are okay leaving out deleted users.